Replicator/tests/ReplicatorSGTest.cc

```diff
@@ -1388,3 +1388,200 @@ TEST_CASE_METHOD(ReplicatorSGTest, "Pinned Certificate Success", "[.SyncServer]"
+        CHECK(keyPath == context->keyPath);
+        return C4SliceResult(UnbreakableEncryption(input, 1));
+    }
+}
+
+static C4SliceResult testDecryptor(void* rawCtx,
+                                   C4String documentID,
+                                   FLDict properties,
+                                   C4String keyPath,
+                                   C4Slice input,
+                                   C4String algorithm,
+                                   C4String keyID,
+                                   C4Error* outError)
+{
+    auto context = (TestEncryptorContext*)rawCtx;
+    context->called++;
+    CHECK(documentID == context->docID);
+    CHECK(keyPath == context->keyPath);
+    return C4SliceResult(UnbreakableEncryption(input, -1));
+}
+
+static C4SliceResult testDecryptorError(void* rawCtx,
+                                        C4String documentID,
+                                        FLDict properties,
+                                        C4String keyPath,
+                                        C4Slice input,
+                                        C4String algorithm,
+                                        C4String keyID,
+                                        C4Error* outError)
+{
+    auto context = (TestEncryptorContext*)((ReplicatorAPITest*)rawCtx)->_decCBContext;
+    if (context->called++ == 0) {
+        *outError = *context->simulateError;
+        return C4SliceResult(nullslice);
+    } else {
+        CHECK(documentID == context->docID);
+        CHECK(keyPath == context->keyPath);
+        return C4SliceResult(UnbreakableEncryption(input, -1));
+    }
+}
+
+TEST_CASE_METHOD(ReplicatorSGTest, "Replicate Encryptor Error", "[.SyncServer]") {
+    _remoteDBName = kScratchDBName;
+    flushScratchDatabase();
+
+    slice originalJSON = R"({"SSN":{"@type":"encryptable","value":"123-45-6789"}})"_sl;
+    slice unencryptedJSON = R"({"ans*wer": 42})"_sl;
+    {
+        TransactionHelper t(db);
+        createFleeceRev(db, "doc01"_sl, kRevID, unencryptedJSON);
+        createFleeceRev(db, "seekrit"_sl, kRevID, originalJSON);
+        createFleeceRev(db, "doc03"_sl, kRevID, unencryptedJSON);
+    }
+
+    TestEncryptorContext encryptContext = {"seekrit", "SSN"};
+    _initParams.propertyEncryptor = &testEncryptorError;
+    _encCBContext = &encryptContext;
+
+    SECTION("LiteCoreDomain, kC4ErrorCrypto") {
+        encryptContext.simulateError = C4Error {LiteCoreDomain, kC4ErrorCrypto};
+        _expectedDocPushErrors = { "seekrit" };
+        replicate(kC4OneShot, kC4Disabled);
+        CHECK(_callbackStatus.progress.documentCount == 2);
+        CHECK(encryptContext.called == 1);
+
+        // Try again with good encryptor and we get the encrypted doc.
+        _initParams.propertyEncryptor = &testEncryptor;
+        _expectedDocPushErrors = {};
+        replicate(kC4OneShot, kC4Disabled);
+        // This is the encrypted doc that we failed to push with the preceding push
```

Comment:
This is not correct. The doc should be skipped after getting kC4ErrorCrypto (permanent) error.

Reply:
taken care